examples/fluids/src/misc.c

```diff
@@ -260,6 +264,18 @@ PetscErrorCode SetBCsFromICs_NS(DM dm, Vec Q, Vec Q_loc) {
                                     "DMPlexInsertBoundaryValues_C", DMPlexInsertBoundaryValues_NS);
   CHKERRQ(ierr);
 
+  PetscCall(DMHasNamedLocalVector(dm, "boundary mask", &boundary_mask_created));
+  if (!boundary_mask_created) {
```

Comment:
We may have to think a bit more about this - In what situation would this mask already exist and be valid for this boundary configuration?

Reply:
I mainly put it there for the dirichlet branch, which currently creates it in a different location. Though they shouldn't be called at the same time and `SetBCsFromICs_NS` would overwrite `DMPlexInsertBoundaryValues` from what the dirichlet function would write anyways.

Reply:
Arguably `SetBCsFromICs` should either be merged in with the Dirichlet function when that gets merged in anyways.

Reply:
Regardless, I'll go ahead and remove that line. I was being a bit over-cautious I think.

Reply:
Removed the if statement.